api/base_controller.go

```diff
@@ -270,13 +302,27 @@ func (c *BaseController) DeleteSingleObject(r *web.Request) (*web.Response, erro
 		},
 		Type:          types.DELETE,
 		State:         types.IN_PROGRESS,
-		ResourceID:    objectID,
+		ResourceID:    resourceID,
 		ResourceType:  c.objectType,
 		PlatformID:    types.SMPlatform,
 		CorrelationID: log.CorrelationIDFromContext(ctx),
-		Context:       c.prepareOperationContextByRequest(r),
+		Context:       opCtx,
+		CascadeRootID: cascadeRootId,
+	}
+	if c.supportsCascadeDelete && opCtx.Cascade {
+		concurrentOp, err := c.scheduler.FindConcurrentOperation(ctx, operation)
```

Comment:
wouldn't it make more sense to first check if there's a concurrent op for the same resource (type + id) and only if not found construct a new one?

Reply:
refactored to check by resource id before creating operation